tests/test_gosymbolextractor.py

```diff
@@ -0,0 +1,52 @@
+import json
+import os
+import shutil
+import tarfile
+
+import pytest
+
+from system.plugins.gosymbolextractor.extractor import GoSymbolExtractor
+from system.helpers.artefact_schema_validator import ArtefactSchemaValidator
+
+
+TESTDATA_DIR = 'testdata'
+
+PATHS = [
+    'github.com/coreos/etcd/b4bddf685b26b4aa70e939445044bdeac822d042',
+    'github.com/eBay/fabio/5e24a09752a75f30448f43773cb529e8ce02ce93',
+    'github.com/PuerkitoBio/agora/fc5914daad197185f2b2fc1873b2084f0e86b6b4',
+    'github.com/hectane/hectane/87062ba605491392423da3e75b1512cfe6350337',
+    'github.com/mozilla-services/heka/149c483a52f0f4c9ea50b73805feebed1b7c45c3',
+    'github.com/pingcap/tidb/482dc3f06c438c320e1fc64ff02a5479d2a989fb',
+    'github.com/cznic/ql/77545ff365ff426dfab06c46d70fa77600cd55af',
+    'github.com/google/cayley/792fe5d42c4dd4d967c3db2e30bd380fcdfb3151',
+    'github.com/influxdata/influxdb/d771eb72f708c523bb2bec35a97f6bb83ae4ef1d',
+    'github.com/google/martian/1afb734cc2eb3a5dab708a901ee07106a94a9186',
+]
+
+
+class TestGoSymbolExtractor(object):
+
+    @pytest.fixture(autouse=True)
+    def prepare(self, request, path):
+        self.path = os.path.join(TESTDATA_DIR, path)
+        def cleanup():
+            shutil.rmtree(self.path, ignore_errors=True)
+        request.addfinalizer(cleanup)
+        cleanup()
```

Comment:
Because tarfile fails to extract its contents if the path already exists. I wanted to make sure that the extracted data are the same as those inside archive.